contracts/contracts/ERC677InitializableBridgeToken.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity 0.4.24;
+
+import "./ERC677InitializableToken.sol";
+
+contract ERC677InitializableBridgeToken is ERC677InitializableToken {
+    mapping(address => bool) internal validBridgeContract;
+
+    function initialize(string _name, string _symbol, uint8 _decimals, address _owner) external initializer {
+        ERC20Mintable.initialize(_owner);
+        ERC20Detailed.initialize(_name, _symbol, _decimals);
+    }
+
+    function addBridgeContract(address _contract) onlyOwner public {
+        require(_contract != address(0) && isContract(_contract));
+        validBridgeContract[_contract] = true;
+    }
+
+    function removeBridgeContract(address _contract) onlyOwner public {
+        require(_contract != address(0) && isContract(_contract));
+        validBridgeContract[_contract] = false;
+    }
+
+    function isBridgeContract(address _contract) public view returns (bool) {
+        return validBridgeContract[_contract];
+    }
+
+    function transferAndCall(address _to, uint _value, bytes _data)
+        external validRecipient(_to) returns (bool)
+    {
+        require(superTransfer(_to, _value));
+        fundReceiver(_to);
+        emit Transfer(msg.sender, _to, _value, _data);
+
+        if (isContract(_to) && !contractFallback(_to, _value, _data)) {
+            if (isBridgeContract(_to)) {
+                revert("reverted here");
+            } else {
+                emit ContractFallbackCallFailed(msg.sender, _to, _value);
+            }
+        }
```

Comment:
I thought we only allow contracts in whitelist as callback. So the code will be:
```
require(isContract(_to) && isBridgeContract(_to) && contractFallback(_to, _value, _data));

```